### ISSUE TITLE
Description of the work done in r20251127

### ISSUE THREAD

**Issue by tomDag25** [ivadomed/moco-dl#21]
The goal of this issue is to describe what was done to train the model in r20251127.

**Comment by tomDag25** [comment]
There are two models : one for dMRI and one for fMRI. 
 
What the repo does : 

1. Generate motion augmeted data ('augmentation .py')
2. Preprocess data ('preprocessing.py')
3. Prepare dataset for training ('dataset_preparation.py')
4. Train DenseNet model ('moco_main.py')
5. Evaluate and test model performance ('test_model.py')

**Comment by tomDag25** [comment]
1 Generate motion augmented data (augmentation.py) 

To generate the data:
`python augmentation.py /path/to/your_data <dmri|fmri>` 

What the code does:  

- uses `config_loader.py` to load `config.yaml` to define the prefixes and subfixes for the files 

- Define a class RandSliceWiseAffine that induces a slice wise shift 
- Parameters are : max_shift (set to 2 here) probability of motion per case, timepoint and slice, the axis to keep unmoved
-  Start with copying the volume (here a volume is a 3d array) 
- Then applies randomly a shift along both axis between ) and 2 voxels

- Define the augment_data function
- Args : input_img, output_img, slicewise_tf
- loads the image data, perfroms augmentation according to slicewise_tf
- Saves the augmenter image in output_img

- Define the main fonction 
- slicewise_tf is defined as    `slicewise_tf = RandSliceWiseAffine(max_shift=2, p_case=0.8, p_frame=0.8, p_slice=0.8, axis=2)`
- Perforns the augmentation defined above on all images. 


Problem : 
Motion correction is applied like : 
```python
tx = np.random.uniform(-self.max_shift, self.max_shift)
ty = np.random.uniform(-self.max_shift, self.max_shift)
out[:, :, d] = affine_transform(
                    out[:, :, d], np.eye(2), offset=(tx, ty), order=0, mode="nearest"
                )
```
This results in no motion correction when tx or ty is between -0.5 and 0.5. 
So probablity of motion in a slice is not 0.8 as thought.

**Comment by tomDag25** [comment]
2 Preprocess data ('preprocessing.py')

- Define process_subject 
- args : target, mode 
- For dMRI: computes the mean, segments the spinal cord on the mean, create a mask along the spinal cord
- For fMRI: computes the mean, segments the spinal cord on the mean, create a mask along the spinal cord
- Then the code creates a target based on the mean (Target is 4d for dMRI and 3d for fMRI)

**Comment by tomDag25** [comment]
3 Prepare dataset for training ('dataset_preparation.py')

- Convert the nii in pytorch tensor (.pt) 
- Create a dataset.json with the file paths 

Problem: no seed and dataset.json provided so the training is not reproducible.

**Comment by tomDag25** [comment]
4 moco_main.py

Trained with a composite loss function cobining 
- Global normalized cross-correlation (GNCC)
- L2 (MSE) and L1 losses inside a mask
- Regularization on translation smoothness across slices and time

Trains a DenseNet.
